### PR TITLE
Switch from clap 2 and structopt to clap 4

### PR DESCRIPTION
**Pull request by onalante-msft**
`structopt` depends on `ansi_term`, which is now being flagged as unmaintained.

<details>
<summary>Tested commands:</summary>
<p>

```
check
check --help
check --verbose
check --output json
check --output text
check --output neither
check --aziot-edged /usr/libexec/aziot/aziot-edged --container-engine-config-file /etc/docker/daemon.json --dont-run 'keyd-running certd-running tpmd-running identityd-running' --expected-aziot-edged-version 1.4.2 --expected-aziot-version 1.4.1 --host unix:///var/run/iotedge/mgmt.sock --iothub-hostname "${IOTHUB_HOSTNAME}" --ntp-server pool.ntp.org:123

check-list
check-list --help

config
config apply
config apply --config-file /etc/aziot/config.toml
config mp --connection-string "${CONNECTION_STRING}"
config mp --connection-string "${CONNECTION_STRING}" --out relpath.toml

list

restart
restart edgeAgent

logs
logs edgeAgent
logs --tail 10 edgeAgent
logs --since '30 seconds' edgeAgent
logs --until '10 seconds' edgeAgent
logs --since '30 seconds' --until '10 seconds' edgeAgent
logs --follow edgeAgent
logs --tail 10 --follow edgeAgent

system
system logs
system logs -- -f
system logs -- --not-a-real-journalctl-option -f
system restart
system stop
system status
system set-log-level
system set-log-level debug
system set-log-level off
system reprovision

support-bundle
support-bundle --output bundle.zip
support-bundle --since '30 seconds' --until '10 seconds'
support-bundle --include-edge-runtime-only

version

help
```

</p>
</details>

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [x] local testing done.

### CODE REVIEW

**Comment by onalante-msft** [comment]
Marked as draft until behavioral testing is done.

**Comment by onalante-msft** [comment]
Behavioral testing done. Regressions found:
- "verbose", "force", and similar flags were expecting values instead of behaving as flags.  Resolved by https://github.com/Azure/iotedge/pull/6720/commits/1a5070af2c0089699dc699c160468653b01d9402.

**Comment by onalante-msft** [comment]
*N.b.* The large diff is due to running `rustfmt` in `api-proxy-module`.

File: edgelet/iotedge/src/main.rs [hunk 1]
```diff
@@ -55,455 +54,480 @@ async fn run() -> anyhow::Result<()> {
...
                 .about("Check for common config and deployment issues")
                 .arg(
-                    Arg::with_name("container-engine-config-file")
+                    Arg::new("container-engine-config-file")
                         .long("container-engine-config-file")
                         .value_name("FILE")
                         .help("Sets the path of the container engine configuration file")
-                        .takes_value(true)
+                        .num_args(1)
+                        .value_parser(clap::value_parser!(OsString))
```
Comment (arsing) [inline edgelet/iotedge/src/main.rs:81]: `s/OsString/PathBuf/` ? Same for the other `OsString`s that are file paths.

**Review by arsing — APPROVED** [review]

File: edgelet/iotedge/src/main.rs (hunk 1 above)
Comment (onalante-msft) [reply edgelet/iotedge/src/main.rs]: bcd8504b8d8ee7b9f0fe6f7402fc5614952b35dc

**Review by arsing — DISMISSED** [review]

**Pull request merged**